python/strings_udf/strings_udf/__init__.py

```diff
@@ -14,8 +14,48 @@
 
 logger = get_logger()
 
-# tracks the version of CUDA used to build the c++ and PTX components
-STRINGS_UDF_PTX_VERSION = (11, 8)
+
+def get_cuda_version_from_ptx_file(path):
+    """
+    https://docs.nvidia.com/cuda/parallel-thread-execution/
+    Each PTX module must begin with a .version
+    directive specifying the PTX language version
+
+    example header:
+    //
+    // Generated by NVIDIA NVVM Compiler
+    //
+    // Compiler Build ID: CL-31057947
+    // Cuda compilation tools, release 11.6, V11.6.124
+    // Based on NVVM 7.0.1
+    //
+
+    .version 7.6
+    .target sm_52
+    .address_size 64
+
+    """
+    with open(path, "r") as f:
+        lines = f.readlines()
+    ver_line = None
+    for line in lines:
+        if ".version" in line:
+            ver_line = line
+            break
+    if ver_line is None:
+        raise ValueError("Could not read cuda version " "from ptx file")
+    version = ver_line.strip("\n").split(" ")[-1]
+    ptx_major, ptx_minor = version.split(".")
+
+    # from ptx_docs/release_notes above:
+    maj_ver_map = {"7": "11", "8": "12"}
+    cuda_major = maj_ver_map.get(ptx_major)
+    if cuda_major is None:
+        raise ValueError(
+            "Could not map major PTX version " f"{ptx_major} to a CUDA version"
+        )
+
+    return (int(cuda_major), int(ptx_minor))
```

Comment:
These parens are extraneous.
```suggestion
    return int(cuda_major), int(ptx_minor)
```